tests/Orange_HRM_Add_Employee.py

```diff
@@ -0,0 +1,91 @@
+import unittest
+from selenium import webdriver
+import time
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.select import Select # works on the tags that have SELECT tag
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+class OrangeHRMEmpAdd(unittest.TestCase):
+
+    def setUp(self) -> None:
+        driver = webdriver.Chrome()
+        driver.maximize_window()
+        self.driver = driver
+        driver.get("http://hrm-online.portnov.com/")
+
+    def tearDown(self) -> None:
+        self.driver.quit()
+
+    def login(self, username, password):
+        # enter username
+        self.driver.find_element_by_id('txtUsername').send_keys(username) if username else None
+
+        if password:
+            # enter password
+            self.driver.find_element_by_id('txtPassword').send_keys(password)
+
+        # click Login button
+        self.driver.find_element_by_id('btnLogin').click()
+
+    def test_emp_add_user(self):
+        driver = self.driver
+        desired_url = 'http://hrm-online.portnov.com/symfony/web/index.php/pim/addEmployee'
+        wait = WebDriverWait(driver, 15)
+        self.login('admin', 'password')
+
+        # Wait provided by Ellie
+
+        wait.until(EC.presence_of_element_located([By.CSS_SELECTOR, '#empsearch_employee_name_empName.inputFormatHint']))
+        driver.find_element_by_id("btnAdd").click()
+        # TODO Need to figure out why the following Waits are not working: #wait.until(EC.url_contains('/viewEmployeeList'))
+        #  #wait.until(EC.element_to_be_clickable([By.ID, "btnAdd"]))
+
+        # Two more Waits for practice
+
+        wait.until(EC.url_to_be(desired_url))
+        wait.until(EC.presence_of_element_located([By.ID, "firstName"]))
+        driver.find_element_by_id("firstName").send_keys("Denis")
+        driver.find_element_by_id("lastName").send_keys("Frolov")
+        employee_id = driver.find_element_by_id("employeeId").get_attribute("value")
+
+        # Uploading a test profile picture
+
+        upload_file = driver.find_element_by_id("photofile")
+        upload_file.send_keys("C:/Users/Denis/Dropbox/My PC (DESKTOP-KJ79GMA)/Documents/Silver Auto Python Bootcamp/upload files/test.jpg")
+        driver.find_element_by_id("chkLogin").click()
+
+        # Waiting for the fields to become visible
+
+        wait.until(EC.visibility_of_element_located([By.ID, "user_name"]))
+        driver.find_element_by_id("user_name").send_keys("Denisfrolov" + employee_id)
+        driver.find_element_by_id("user_password").send_keys("password")
+        driver.find_element_by_id("re_password").send_keys("password")
+        driver.find_element_by_id("status").click()
+        driver.find_element_by_id("btnSave").click()
+
+        # 2 Waits for practice
+
+        wait.until(EC.url_changes)
+        wait.until((EC.presence_of_element_located([By.ID, "btnSave"])))
```

Comment:
Actually this same locator is present on both pages, so this wait wouldn’t be helpful

Reply:
Ahh, didn't notice that, thanks. Changed it to this:

        # Making sure Personal Details banner is displayed
        wait.until((EC.presence_of_element_located([By.XPATH, "//h1[contains(text(),'Personal Details')]"])))